pkg/image/unprivileged/resolver.go

```diff
@@ -0,0 +1,51 @@
+package unprivileged
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/containerd/containerd/remotes"
+	"github.com/containerd/containerd/remotes/docker"
+	auth "github.com/deislabs/oras/pkg/auth/docker"
+)
+
+func NewResolver(username, password string, configs ...string) (remotes.Resolver, error) {
+	// TODO(njhale): make plainHTTP and insecure optional
+	plainHTTP := true
```

Comment:
We should figure out these options before we merge - not supporting tls verification will be an issue.